corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js

```diff
@@ -420,16 +429,18 @@ FormplayerFrontend.on('view:phone', function() {
  * to complete progress.
  */
 FormplayerFrontend.on('clearProgress', function() {
-    var progressView = FormplayerFrontend.regions.loadingProgress.currentView,
-        progressFinishTimeout = 0;
+    var progressView = FormplayerFrontend.regions.loadingProgress.currentView;
+
     if (progressView) {
-        progressFinishTimeout = 200;
+        var progressFinishTimeout = 200;
         progressView.setProgress(1, progressFinishTimeout);
-    }
 
-    setTimeout(function() {
+        setTimeout(function() {
```

Comment:
This is the fix for the progress view getting wiped out immediately after I was showing it. Apparently even when the `progressFinishTimeout` is set to 0, making a call via `setTimeout()` causes enough of a delay that `FormplayerFrontend.regions.loadingProgress.empty()` wasn't getting called immediately. So this change makes it so that we just don't use the `setTimeout()` paradigm at all unless it's strictly necessary.